#! /bin/sh /usr/share/dpatch/dpatch-run
## 01sysexit.dpatch by  <az@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: upstream #25379: older pythons error-out inappropriately

@DPATCH@
diff -urNad duplicity-0.5.06~/duplicity duplicity-0.5.06/duplicity
--- duplicity-0.5.06~/duplicity	2009-01-10 00:05:11.000000000 +1000
+++ duplicity-0.5.06/duplicity	2009-01-31 14:26:02.565314440 +1000
@@ -589,6 +589,8 @@
         log.FatalError("%s: %s" % (e.__class__.__name__, str(e)),
                        log.ErrorCode.user_error,
                        e.__class__.__name__)
+    except SystemExit, e:
+        pass
     except Exception, e:
         log.FatalError("%s" % (''.join(traceback.format_exception(*sys.exc_info()))),
                        log.ErrorCode.exception,
